libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -1160,6 +1184,39 @@ internal class MapRouteLine(
                 it.addStringProperty(WAYPOINT_PROPERTY_KEY, propValue)
             }
         }
+
+        fun findDistanceOfPointAlongLine(line: LineString, point: Point): Double {
+            val linePoints = line.coordinates()
+            val runningTotalDistance = mutableListOf<Double>()
+            var distanceToReturn = 0.0
+            for (index in 0 until linePoints.size) {
+                when (index == 0) {
+                    true -> 0.0
+                    false -> TurfMeasurement.distance(linePoints[index - 1], linePoints[index], TurfConstants.UNIT_METERS)
+                }.also {
+                    runningTotalDistance.add(it)
+                }
+
+                val distanceToPoint = TurfMeasurement.distance(linePoints[index], point, TurfConstants.UNIT_METERS)
+                val distanceToNextIndex = if ((index + 1) < linePoints.size) {
+                    TurfMeasurement.distance(
+                        linePoints[index],
+                        linePoints[index + 1],
+                        TurfConstants.UNIT_METERS
+                    )
```

Comment:
This calculation starts overlapping with the one done in the `when` clause above after an initial couple of loops and we're duplicating the work. Maybe incrementing by 2 instead of 1, calculating and comparing all distances at once will allow us to avoid this duplication.

Reply:
I changed the calculation.